pyanaconda/ui/gui/spokes/lib/installation_source_helpers.py

```diff
@@ -163,6 +163,74 @@ def check_duplicate_repo_names(repo_names):
     return InputCheck.CHECK_OK
 
 
+def validate_repo_url(protocol, path, repo_name=None):
+    """Validate a repo URL.
+
+    :param protocol: a string with a protocol ("http")
+    :param path: a string with a path ("server/")
+    :param repo_name: a name of the repository or None
+    :return: an error message or None
+    """
+    # If this is HTTP/HTTPS/FTP, use the URL_PARSE regex
+    if protocol in (PROTOCOL_HTTP, PROTOCOL_HTTPS, PROTOCOL_FTP):
+        if not path:
+            if repo_name:
+                return _("Repository %s has empty url") % repo_name
+            else:
+                return _("URL is empty")
```

Comment:
I think it could be better to also split the helper into two for repo and non-repo, if you're already refactoring. Having `if repo_name` for each message is weird.

Reply:
I would even split based on the top-level `if-elif-else` it's pretty hard to follow this way.